File: difficult/Q4/arraydifference.py
def array_difference(list1, list2):
    """ Function to return a list of elements from {list1} that aren't in {list2} """
    return [num for num in list1 if num not in list2]


print(array_difference([1, 2], [2]) == [1]) # True
print(array_difference([1, 2, 3], [1, 8, 3]) == [2]) # True
print(array_difference([], [3, 4]) == []) # True
#

File: tough/Q4/arraydifference.js
// Function to return a list of elements from {arr1} that aren't in {arr2}
const arrayDifference = (arr1, arr2) =>
  arr1.filter((item) => !arr2.includes(item));

console.log(
  JSON.stringify(arrayDifference([1, 2], [2])) === JSON.stringify([1])
); // true
console.log(
  JSON.stringify(arrayDifference([1, 2, 3], [1, 8, 3])) === JSON.stringify([2])
); // true
console.log(JSON.stringify(arrayDifference([], [3, 4])) === JSON.stringify([])); // true
